src/query.c

```diff
@@ -1065,6 +1069,10 @@ static IndexIterator *Query_EvalTagLexRangeNode(QueryEvalCtx *q, TagIndex *idx,
   if (ctx.nits == 0) {
     rm_free(ctx.its);
     return NULL;
+  } else if (ctx.nits == 1) {
+    IndexIterator *ret = ctx.its[0];
+    rm_free(ctx.its);
```

Comment:
Need to nullify `ctx.its` after freeing it (same for the rest of the changes)? Basically, is anyone asking `if (ctx.its)`?